extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -333,13 +333,28 @@ fn generate_markdown_from_item_data(
     if let Some(sig) = &doc_item.signature() {
         if !sig.is_empty() {
             // TODO(#1525) add cairo support to mdbook
-            writeln!(&mut markdown, "```rust\n{sig}\n```\n")?;
+            let signature_formatted = format!(
+                "<pre><code class=\"language-rust\">{}</code></pre>\n",
+                escape_html_markers(sig)
+            );
+            writeln!(&mut markdown, "{}", signature_formatted)?;
```

Comment:
Why not 
```suggestion
            writeln!(&mut markdown, "<pre><code class=\"language-rust\">{}</code></pre>\n", escape_html_markers(sig))?;
```
You should be able to handle formatting in writeln directly:
https://doc.rust-lang.org/std/macro.writeln.html